admindb/sqlite/invites.go

```diff
@@ -0,0 +1,207 @@
+package sqlite
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/friendsofgo/errors"
+	"github.com/mattn/go-sqlite3"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"github.com/ssb-ngi-pointer/go-ssb-room/admindb"
+	"github.com/ssb-ngi-pointer/go-ssb-room/admindb/sqlite/models"
+	refs "go.mindeco.de/ssb-refs"
+)
+
+// make sure to implement interfaces correctly
+var _ admindb.InviteService = (*Invites)(nil)
+
+// Invites implements the admindb.InviteService.
+// Tokens are stored hashed with sha256 on disk to protect against someone who gaining read-access on the database.
+type Invites struct {
+	db *sql.DB
+
+	allowList *AllowList
+}
+
+const tokenLength = 50
```

Comment:
magic number? or why is it 50?

Reply:
The rooms2 doc leaves this open. cc @staltz

Reply:
What's the size of normal pub invite codes? Should be the same as this one too, no? Is this something that needs to be spec'd? (I don't see why all the rooms would have to have the same lengths)

Reply:
Same for the choice of encoding the tokens. url safe base64 seemed like a better choice then hex.

Also not super sure it needs to be defined strictly since it just needs to be returned and passed back as an opaque string.

Reply:
> What's the size of normal pub invite codes? 

In classic ssb-invites it's a seed for a keypair for ed22519, which is 32bytes. But that protocol is not susceptible to brute forcing an HTTP endpoint, which is why I made it longer.